### ISSUE TITLE
No items fetched after fetching less than invisibleItemsThreshold items

### ISSUE THREAD

**Issue by KrisBraun** [EdsonBueno/infinite_scroll_pagination#311]
Hit this issue when using `invisibleItemsThreshold = 20` with a fetcher that returns one item at a time.

[This code](https://github.com/EdsonBueno/infinite_scroll_pagination/blob/master/lib/src/widgets/helpers/paged_layout_builder.dart#L267) assumes a fetch only needs to be triggered when crossing the threshold, while further fetches may be required after that one completes.

**Comment by caiombueno** [comment]
Hi @KrisBraun. A page is only requested when the trigger index item (`newPageRequestTriggerIndex`) is built/rebuilt. When the `invisibleItemsThreshold` value is too high, the trigger index item won't be at the end of the list. So, when you reach the end of the list, no more items will be fetched.

**Comment by KrisBraun** [comment]
Right. My use case is that I'd like to prefetch items early at an offset greater than the fetch page size (which is non-deterministic in my case). 

If there's a way to detect when new items are added, this could be handled by generating another fetch if the total remaining items is still below the threshold.

**Comment by clragon** [comment]
You can listen to the pagecontroller, which will send a notification when new items are added.
Is that an appropriate solution for your issue?

**Comment by KrisBraun** [comment]
I think for that to work, I'd need a way to determine the number of invisible items given the current position, so I could trigger additional fetches if there are fewer than the threshold remaining. As far as I can tell, none of that is accessible through the PagingController.

The cleanest solution from my perspective (external to the implementation) would be for the PagingController to fetch additional pages until it is above the threshold.

**Comment by clragon** [comment]
This seems reasonable on the first glance 
but I am unsure if it is desirable for the package to start making many requests like that.

I dont have time to specify this right now, but
- Does this align with what we want?
- Would a user be surprised by this behaviour?
- Is there a situation where this would have negative effects?

**Comment by xsahil03x** [comment]
@KrisBraun you can use [super_paging](https://pub.dev/packages/super_paging) for your usecase.

**Comment by clragon** [comment]
I have been able to reproduce this issue. Now the important question is, how do we fix it?

I would have assumed that the children of the layout rebuild alongside it when its state changes, which would prompt it to get another call to request a new page. But it seems like that is not the case. I need to investigate this a bit further.

**Comment by clragon** [comment]
I have deployed a fix for this issue that I believe will work for v5. 

Instead of only requesting a page when we reach the trigger exactly, we will request a page should we build past the trigger.
This prevents large thresholds from being calculated behind the currently to-be-built item, making it impossible to trigger a new page.

I hope this works for you. Thanks!
